game_patch/object/entity.cpp

```diff
@@ -201,6 +202,79 @@ CodeInjection entity_process_pre_hide_riot_shield_injection{
     },
 };
 
+FunHook<void(int)> entity_blood_throw_gibs_hook{
+    0x0042E3C0, [](int handle) {
+        if (g_game_config.gibbing && !rf::is_multi) {
+            rf::Object* objp = rf::obj_from_handle(handle);
+
+            if (!objp) {
+                return; // invalid object
+            }
+
+            rf::Vector3 pos = objp->pos;
+            rf::GRoom* room = objp->room;
+            int explosion_vclip = 30;
+            int chunk_explosion_vclip = 30;
+            float explosion_vclip_radius = 1.0f;
+            rf::String debris_filename = "df_meatchunks0.V3D";
+
+            int debris_max_lifetime = 7000;
+            float debris_velocity = 8.5f;
+            float damage_scale = 1.0f;
+            rf::String cust_snd_set = "gib bounce";
+
+            if (objp->type == rf::OT_ENTITY) {
+                rf::Entity* ep = static_cast<rf::Entity*>(objp);
+
+                explosion_vclip = (ep->info->explode_vclip_index < 0)
+					? ep->info->explode_vclip_index : explosion_vclip;
+
+                explosion_vclip_radius = (ep->info->explode_vclip_radius > 0.0f)
+					? ep->info->explode_vclip_radius : explosion_vclip_radius;
+
+                debris_filename = (!ep->info->debris_filename.empty())
+					? ep->info->debris_filename : debris_filename;
+            }
+            else if (objp->type = rf::OT_CORPSE) {
+            }
+            else {
+                return;
+            }
+
+            rf::game_do_explosion(
+                explosion_vclip, room, 0, &pos, explosion_vclip_radius, damage_scale, 0);
+
+            rf::debris_spawn_from_object(
+                objp, debris_filename, chunk_explosion_vclip, debris_max_lifetime, debris_velocity, &cust_snd_set);
+        }
+    }
+};
+
+CodeInjection entity_damage_explosive_death_injection{
+    0x0041A413,
+    [](auto& regs) {        
+        rf::Entity* ep = regs.esi;
+        if (g_game_config.gibbing && !rf::is_multi && ep) {
+            int damage_type = regs.ebp;
+            ep->ai.explosive_last_damage = (damage_type == 3);
+            if (ep->ai.explosive_last_damage && ep->info->body_temp >= 90.0f) {
+                ep->death_anim_index = -1;
+            }
+        }
+    }
+};
+
+CodeInjection entity_dying_frame_explode_injection{
+    0x0041EE4C,
+    [](auto& regs) {
+        rf::Entity* ep = regs.esi;
+        if (g_game_config.gibbing && !rf::is_multi && ep && ep->ai.explosive_last_damage &&
```

Comment:
you repeat all the conditions here... Please follow DRY rule (Don't Repeat Yourself). 
But still I'm not sure it should be checked in two places. 
I can see that RF checks for flag 0x80, maybe you could set it in the other place :thinking: Maybe it's not used anywhere and Volition originally used it to pass the information that entity should explode with gibbing. Using it could also remove the need for adding `explosive_last_damage`